src/pages/chainlink-functions/resources/add-functions-to-projects.md

```diff
@@ -0,0 +1,317 @@
+
+In general, a consumer contract requires several components:
+
+<ChainlinkFunctions section="functions-consumer" />
+
+Next, deploy the contract.
+
+### Deploy a consumer contract
+
+You can deploy consumer contracts using the [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit), but this example shows how to deploy the contract programmatically using a script.
+
+1. Create a script named `deploy.js` in the `./scripts` folder of your project:
+
+   ```shell
+   touch ./scripts/deploy.js && open ./scripts/deploy.js
+   ```
+
+1. Copy the following code into the script.
+
+   <CodeSample src="samples/Functions/adding-functions-to-projects/deploy.js" />
+
+1. Set the `oracleAddress` to the oracle address on the network that you want to use. Each network has a unique DON with a unique oracle address. See the [Supported Networks](/chainlink-functions/supported-networks) page for a list of supported networks and oracle addresses. For this example, use the address for the oracle on Polygon Mumbai:
+
+   ```javascript
+   const oracleAddress = "0xeA6721aC65BCeD841B8ec3fc5fEdeA6141a0aDE4"
+   ```
+
+1. Set `const contractName` to the name of the contract that you want Hardhat to deploy. The `ethers.getContractFactory(contractName);` line creates a `ContractFactory` object using the contract that you define. For this example, use `FunctionsConsumer`.
+
+   ```javascript
+   const contractName = "FunctionsConsumer"
+
+   ⋮
+
+   const consumerContract = await ethers.getContractFactory(contractName);
+   ```
+
+1. Save the file and run the script to deploy your contract.
+
+   ```shell
+   npx hardhat run ./scripts/deploy.js
+   ```
+
+1. If your contract deployed successfully, you will see the deployed consumer address. Record this address to use later:
+
+   ```text
+   Deployed Functions Consumer address: 0xE83188aDF26bd6c18076704fC8EC68CBda454f76
+   ```
+
+Next, create and fund your Chainlink Functions subscription.
+
+### Create and fund a subscription
+
+You can use the [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit) to create and manage your subscriptions. See [Managing Subscriptions](/chainlink-functions/resources/subscriptions) for instructions.
+
+This example shows how to create and manage subscriptions programmatically. You can create the subscription, fund the subscription, and authorize the consumer all in one script.
+
+1. Ensure that the wallet address you are using to create the subscription has a sufficient LINK balance. You can get testnet LINK at [faucets.chain.link](https://faucets.chain.link/mumbai). To get testnet funds for other networks, see the [LINK Token Contracts](/resources/link-token-contracts) page.
```

Comment:
do we also need the discaimer about being signed up for the whitelist here?

Reply:
Added a reminder here and a notice at the top of the page. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0